code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -318,66 +318,22 @@
 			LL_dir.icon_state = "trackoff"
 			return
 
-	if(isxeno(tracked))
-		var/mob/living/carbon/xenomorph/xeno_tracked = tracked
-		if(QDELETED(xeno_tracked))
-			tracked = null
-			return
-		if(xeno_tracked == src) // No need to track ourselves
-			LL_dir.icon_state = "trackoff"
-			return
-		if(xeno_tracked.z != z || get_dist(src, xeno_tracked) < 1)
-			LL_dir.icon_state = "trackondirect"
-			return
-		var/area/A = get_area(loc)
-		var/area/QA = get_area(xeno_tracked.loc)
-		if(A.fake_zlevel == QA.fake_zlevel)
-			LL_dir.icon_state = "trackon"
-			LL_dir.setDir(get_dir(src, xeno_tracked))
-			return
-
-		LL_dir.icon_state = "trackondirect"
+	if(tracked == src) // No need to track ourselves
+		LL_dir.icon_state = "trackoff"
 		return
-
-	if(isresinsilo(tracked))
-		var/mob/living/carbon/xenomorph/silo_tracked = tracked
-		if(QDELETED(silo_tracked))
-			tracked = null
-			return
-		if(silo_tracked.z != z || get_dist(src, silo_tracked) < 1)
-			LL_dir.icon_state = "trackondirect"
-			return
-
-		var/area/A = get_area(loc)
-		var/area/QA = get_area(silo_tracked.loc)
-		if(A.fake_zlevel == QA.fake_zlevel)
-			LL_dir.icon_state = "trackon"
-			LL_dir.setDir(get_dir(src, silo_tracked))
-			return
+	if(tracked.z != z || get_dist(src, tracked) < 1)
 		LL_dir.icon_state = "trackondirect"
 		return
-
-	if(istype(tracked, /obj/machinery/nuclearbomb))
-		var/obj/machinery/nuclearbomb/nuke_tracked = tracked
-		if(QDELETED(nuke_tracked))
-			tracked = null
-			return
-		if(!nuke_tracked.timer_enabled)
-			LL_dir.icon_state = "trackoff"
-			return
-		if(nuke_tracked.z != z || get_dist(src, nuke_tracked) < 1)
-			LL_dir.icon_state = "trackondirect"
-			return
-		var/area/A = get_area(loc)
-		var/area/QA = get_area(nuke_tracked.loc)
-		if(A.fake_zlevel == QA.fake_zlevel)
-			LL_dir.icon_state = "trackon"
-			LL_dir.setDir(get_dir(src, nuke_tracked))
-			return
-
-		LL_dir.icon_state = "trackondirect"
+	var/area/A = get_area(loc)
+	var/area/QA = get_area(tracked.loc)
+	if(A.fake_zlevel == QA.fake_zlevel)
+		LL_dir.icon_state = "trackon"
+		LL_dir.setDir(get_dir(src, tracked))
 		return
 
+	LL_dir.icon_state = "trackondirect"
+	return
```

Comment:
hm ?

Reply:
this was the same code over and over.

Since i need to be able to track the king pod, I needed a small refactor